src/inmemory_cache.rs

```diff
@@ -0,0 +1,24 @@
+use std::sync::Arc;
+
+use crate::{
+    block::Block, db_state::SsTableId, filter::BloomFilter, flatbuffer_types::SsTableIndexOwned,
+};
+
+#[derive(Clone)]
+#[allow(dead_code)]
+pub(crate) enum CachedBlock {
+    Block(Arc<Block>),
+    Index(Arc<SsTableIndexOwned>),
+    Filter(Arc<BloomFilter>),
+}
+
+pub(crate) type BlockCache = moka::future::Cache<(SsTableId, usize), CachedBlock>;
```

Comment:
One other note: @rodesai requested that we put the moka cache behind a trait ([here](https://github.com/slatedb/slatedb/pull/137#discussion_r1728397369)), that way we don't back the moka API into our cache API.

Some kind of `trait Cache` with some very basic methods. We can implement a MokaCache and construct the moka Cache accordingly.

Reply:
Updated to use a trait for Cache